packages/cluster/src/common.js

```diff
@@ -99,9 +119,28 @@ function resolveConfiguration(argv = {}, cache = {}, config = {}) {
 
     reporter.debug('Resolved configuration\n', resolved)
 
+    await argv.getCache().write(
+        file,
+        JSON.stringify(
+            {
+                channel: resolved.channel,
+                dbVersion: resolved.dbVersion,
+                dhis2Version: resolved.dhis2Version,
+                customContext: resolved.customContext,
+                image: resolved.image,
+                port: resolved.port,
+            },
+            null,
+            4
+        )
+    )
+
     return resolved
 }
 
+module.exports.cleanCache = async argv =>
+    await argv.getCache().purge(path.join(clusterDir, argv.name))
```

Comment:
Why doesn't this use the `({ name })` object parameter deconstruction form?  I find `argv` arguments opaque unless there are potentially unknown or dynamic properties.

Reply:
I thought about it, but I thought the other way around, that it makes it less clear what is being used from where.

Changed in 4f7d7d2